pkg/collector/corechecks/servicediscovery/apm/detect.go

```diff
@@ -200,6 +200,8 @@ func nodeDetector(ctx usm.DetectionContext) Instrumentation {
 	return None
 }
 
+var javaAgentRegex = regexp.MustCompile("-javaagent:.*(?:datadog|dd-java-agent)")
```

Comment:
Should also handle `dd-trace-agent.jar`
https://ddstaging.datadoghq.com/process?query=dd-trace-agent&selectedTopGraph=none